File: Python/Caesar Algorithm/Caesar.py
# Caesar Algorithm

# global variable
A = "" # new message
chs = 0 # user choise variable
key = 0 # key -> (positive number)

# ----------
# menu
def menu():
    global chs 
    print("1. Decode")
    print("2. Encode")
    print("3. Quit")
    
    while True:
        chs = int(input("Please enter your choise ? : ")) # <class 'int'>
        if ((chs == 1)or(chs == 2)or(chs == 3)):
            break;

# ----------
# decode (decipher)
def decode():
    global A
    global key
    msg = input("Please enter your message : ") # <class 'str'>
    msg = msg.upper() # MESSAGE
    while True:
        key = int(input("Please enter your key : "))
        if key > 0:
            break
        
    A = ""
    for i in msg: # decipher
        temp = 0
        if ord(i)>64 and ord(i) < (65+key):
            temp = (ord(i)-(key))
            temp = 64-temp
            temp = 90 - temp
            A += chr(temp)
        elif ord(i) < 65:
            A += " "
        else:
            A += chr(ord(i)-key)
    print("Result :",A) # Result : YUNUS

# ----------
# encode (cipher)
def encode():
    global A
    global key
    msg = input("Please enter your message : ") # <class 'str'>
    msg = msg.upper() # MESSAGE
    while True:
        key = int(input("Please enter your key : "))
        if key > 0:
            break
    A = ""
    for i in msg: # cipher
        temp = 0
        if ord(i) > (90-key):
            temp = (ord(i)+(key))%90
            temp = temp + 64
            A += chr(temp)
        elif ord(i) < 65:
            A += " " 
        else:
            A += chr(ord(i)+key)
    print("Result :",A) # Result : DZSZX


# main function
while True:
    menu()
    if(chs == 1): # decipher
        decode()
    elif(chs == 2): # cipher
        encode()
    elif(chs == 3): # quit
        break
    
    
    
    
    
    
    
    

File: C/Caesar Algorithm/Caesar.c
// Caesar Algorithm

#include <stdio.h>
#include <conio.h>
#include <ctype.h>
#include <stdlib.h>

int main()
{
    char msg[40];
    int i=0, j=0, length=0;
    char A[length], ch;
    int key;

    // input message
    printf("Please enter your message : ");
    gets(msg);

    // uppercase
    while(msg[j]){

        ch = msg[j];
        msg[j] = toupper(ch);
        j++;
    }
    while(msg[i] != '\0')  // string length
    {
        length++;
        i++;
    }

    // input key
    printf("Please enter your key number : ");
    while(1){
        scanf("%d",&key);
        if((key>0)&&(key<27)){
            break;
        }
    }

    // new message
    for (i=0; i<length; i++)
    {
        int B=0;
        B=(int)msg[i];
        int temp = 0;
        if (B > (90-key))
        {
            temp = (B+key)%90;
            temp = temp+64;
            printf("%c",(char)temp);
            A[i] += (char)temp;
        }
        else
        {
            printf("%c",(char)(B+key));
            A[i] += (char)(B+key);
        }
    }

    getch();
    return 0;
}
